code/modules/mob/living/simple_animal/hostile/regalrat.dm

```diff
@@ -109,55 +156,38 @@
 	else
 		to_chat(src, "<span class='warning'>You feel fine, no need to eat anything!</span>")
 
-
-/mob/living/simple_animal/hostile/regalrat/controlled
-	name = "regal rat"
-
 /mob/living/simple_animal/hostile/regalrat/controlled/Initialize()
 	. = ..()
 	INVOKE_ASYNC(src, .proc/get_player)
+	var/kingdom = pick("Plague","Miasma","Maintenance","Trash","Garbage","Rat","Vermin","Cheese")
+	var/title = pick("King","Lord","Prince","Emperor","Supreme","Overlord","Master","Shogun","Bojar","Tsar")
+	name = kingdom + " " + title
 
 
 /**
- *This action creates trash, money, dirt, and cheese.
+ *Increase the rat king's domain
  */
-/datum/action/cooldown/coffer
-	name = "Fill Coffers"
-	desc = "Your newly granted regality and poise let you scavenge for lost junk, but more importantly, cheese."
+
+/datum/action/cooldown/domain
+	name = "Rat King's Domain"
+	desc = "Corrupts this area to be more suitable for your rat army."
+	cooldown_time = 6 SECONDS
 	icon_icon = 'icons/mob/actions/actions_animal.dmi'
 	background_icon_state = "bg_clock"
 	button_icon_state = "coffer"
-	cooldown_time = 50
 
-/datum/action/cooldown/coffer/Trigger()
+/datum/action/cooldown/domain/Trigger()
 	. = ..()
-	if(!.)
-		return
 	var/turf/T = get_turf(owner)
-	var/loot = rand(1,100)
-	switch(loot)
-		if(1 to 5)
-			to_chat(owner, "<span class='notice'>Score! You find some cheese!</span>")
-			new /obj/item/food/cheesewedge(T)
-		if(6 to 10)
-			var/pickedcoin = pick(GLOB.ratking_coins)
-			to_chat(owner, "<span class='notice'>You find some leftover coins. More for the royal treasury!</span>")
-			for(var/i = 1 to rand(1,3))
-				new pickedcoin(T)
-		if(11)
-			to_chat(owner, "<span class='notice'>You find a... Hunh. This coin doesn't look right.</span>")
-			var/rarecoin = rand(1,2)
-			if (rarecoin == 1)
-				new /obj/item/coin/twoheaded(T)
-			else
-				new /obj/item/coin/antagtoken(T)
-		if(12 to 40)
-			var/pickedtrash = pick(GLOB.ratking_trash)
-			to_chat(owner, "<span class='notice'>You just find more garbage and dirt. Lovely, but beneath you now.</span>")
-			new /obj/effect/decal/cleanable/dirt(T)
-			new pickedtrash(T)
-		if(41 to 100)
-			to_chat(owner, "<span class='notice'>Drat. Nothing.</span>")
+	T.atmos_spawn_air("miasma=4;TEMP=[T20C]")
```

Comment:
@LemonInTheDark is this an issue you reckon?

Reply:
For the record, it's not miasma itself that was its issue, it was the constant generation of it that made turfs never settle.

Reply:
For the record, it's not miasma itself that was its issue, it was the constant generation of it that made turfs never settle.

Reply:
The 6 second cooldown is a bit sus

Reply:
> The 6 second cooldown is a bit sus

What do you mean?

It generates a bit of miasma, but not enough to fill a tile with the required miasma threshold to trigger the regeneration on the rat.

This ability is intended to work in a way like the xeno resin weeds; use it multiple times in a room to fill it in miasma and allow the regal rat to regenerate.

Reply:
To be honest with you I'm kinda tempted to remove miasma at this point, since it's not serving the purpose it was built to. Keep in mind that the 4 miasma this would spawn will dissipate into the area the rat is in, so it'll take a while for this to become fully visible. I think this should be enough to cause disease though.

Reply:
If you can give it purpose or meaning I've got no quarrel tho, as things sit on master it's kinda just useless.

Reply:
>Keep in mind that the 4 miasma this would spawn will dissipate into the area the rat is in, so it'll take a while for this to become fully visible.

This ability is intended to be used in maints tunnels/small maints rooms to fill them with miasma, where the king rat will regenerate health.